22579 - Add FED Column & Tooltips (continuation table)

ArwenQin opened the pull request:
*Issue #:*
https://github.com/bcgov/entity/issues/22579

*Description of changes:*
- Added Future Effective Date column to table
- Added red/orange tooltips to future effective date
- Added red/orange tooltips to NR number when it has expired /<14 days
- Added action for NR redirect to Namex UI
- Added hover text showing what statuses are selected
 - For date pickers: show the date range truncated and the full text on hover
 - Added close icon to all search fields
 - Fixed the bug: calendar icon can't open the date picker

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

ArwenQin commented:
Updated the Table name, FED column and actions icon styles:
![image](https://github.com/user-attachments/assets/9667b4ae-4358-478f-9457-6f87d302fc1b)

severinbeauvais commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://bcregistry-account-dev--pr-2978-f8g8fzgr.web.app

severinbeauvais commented:
I think you should make the Status column wider so it fits without horizontal scrolling. You can make some other columns narrower, such as the dates and the NR number.

severinbeauvais commented:
Ask Janis if the Name Examination should open in a new page (tab).

seeker25 commented on auth-web/src/components/auth/staff/continuation-application/ContinuationApplicationTable.vue:
```diff
@@ -153,9 +282,27 @@
+                              <span>
+                                <!-- when hover over 'Multiple Selected' showing what are selected. -->
+                                {{
+                                  reviewParams[header.value]
```
computed?

ArwenQin replied:
Updated

severinbeauvais replied:
Fixed. Resolving.

seeker25 reviewed (approved):
LGTM, need entities review though

severinbeauvais commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://bcregistry-account-dev--pr-2978-f8g8fzgr.web.app

ArwenQin commented:
> Temporary Url for review: https://bcregistry-account-dev--pr-2978-f8g8fzgr.web.app

This link is not the current version, could you pleaser comment /gcbrun again? @severinbeauvais  Thanks!

severinbeauvais commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://bcregistry-account-dev--pr-2978-f8g8fzgr.web.app

severinbeauvais commented:
> Updated, made the NR number, date and identifier column narrower. Updated the redirect to NameX, will open in a new tab. 

Looks great!

Since there's an 'X' to clear the dates and Status, what do you think of adding an 'X' to clear the NR Number, Identifying Number and Completing Party?

![image](https://github.com/user-attachments/assets/76cfb244-1f61-4527-ad88-df638fe35324)

severinbeauvais commented:
PS You need to "pull up" the icons in the header by 1 or 2 px.

![image](https://github.com/user-attachments/assets/3e3dcc83-1722-4247-8c4d-93ca5f481af7)

![image](https://github.com/user-attachments/assets/99b74edb-d93a-442a-a8eb-b5f250232289)

Also, can you check what the dropdown menu is attached to? It obscures its action button so I have to click away from it to close it.

![image](https://github.com/user-attachments/assets/293f919b-90bd-4e22-b4ca-449cc87111a7)

severinbeauvais commented:
Do you think we should be able to open the NR when the application is approved or rejected?

![image](https://github.com/user-attachments/assets/b7408afa-d2c5-4df8-87a0-146d417a3c5d)

![image](https://github.com/user-attachments/assets/b0f6e716-b35a-4429-ad45-bb2886bec14e)

I'm thinking "yes" but check with Janis and/or Mihai.

ArwenQin commented:
> Do you think we should be able to open the NR when the application is approved or rejected?

Do you mean "open Name Request" action? 
I saw a Note "that NRs for Approved/Rejected/Abandoned reviews will not have an expiry date or id or phone/email, so do not show the "open name request" action for these ones."

ArwenQin commented:
> PS You need to "pull up" the icons in the header by 1 or 2 px.
> 
> ![image](https://private-user-images.githubusercontent.com/10002889/360599786-3e3dcc83-1722-4247-8c4d-93ca5f481af7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzNTQ5MjIsIm5iZiI6MTcyNDM1NDYyMiwicGF0aCI6Ii8xMDAwMjg4OS8zNjA1OTk3ODYtM2UzZGNjODMtMTcyMi00MjQ3LThjNGQtOTNjYTVmNDgxYWY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIyVDE5MjM0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk3NDcyYjhkNjA4Y2UwYWIxMTI3NjU4ZjUzNzVlNDlhNzIzNTM0ZGU2NGNhNDkwODU0ODNlZDNiMjU4ZDk2NDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.uEj2Pj5ewvS0j3GCU0m8eTWBVeNSLN7OGwrt6z7eFRQ)
> 
> ![image](https://private-user-images.githubusercontent.com/10002889/360599847-99b74edb-d93a-442a-a8eb-b5f250232289.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzNTQ5MjIsIm5iZiI6MTcyNDM1NDYyMiwicGF0aCI6Ii8xMDAwMjg4OS8zNjA1OTk4NDctOTliNzRlZGItZDkzYS00NDJhLWE4ZWItYjVmMjUwMjMyMjg5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIyVDE5MjM0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg3MTVhY2U2ZTg2NjAwNjgzOTMzMmM2Mjc1MWIwYjRkMWU3MGY1YzhiMTE5YTI3ODAwMDdkYWI1ZGZjMDYwNzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.M4BYDMXmCM3JrXqiHj9vhGUMf5wuXHTs8_KDEpBvGkA)
> 
> Also, can you check what the dropdown menu is attached to? It obscures its action button so I have to click away from it to close it.
> 
> ![image](https://private-user-images.githubusercontent.com/10002889/360600111-293f919b-90bd-4e22-b4ca-449cc87111a7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzNTQ5MjIsIm5iZiI6MTcyNDM1NDYyMiwicGF0aCI6Ii8xMDAwMjg4OS8zNjA2MDAxMTEtMjkzZjkxOWItOTBiZC00ZTIyLWI0Y2EtNDQ5Y2M4NzExMWE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIyVDE5MjM0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUyODIzZWQ5NzEzMTliNDY1NmJjZDc5ODEyOTgzNmFkNWYzYzYyNzYzOWU1YmU2OWQxZDBlYjhmNjZjZTlkNDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7TTCIT2khlPNYRXkpWvVcADTbTPit-wy26I2Sumk1gI)

It's attached to the second row header <th>

severinbeauvais commented on auth-web/src/components/auth/staff/continuation-application/ContinuationApplicationTable.vue:
```diff
@@ -190,13 +343,41 @@
+                        </v-list-item>
+                      </v-list>
+                    </v-menu>
+                  </span>
```
I found a bug: the sub-menu opens multiple times when I click the down arrow to open it. Double-check this code, or maybe it needs a 'key' prop?.

![image](https://github.com/user-attachments/assets/e4b501f3-4e8f-4020-83cb-6b081b102fac)

severinbeauvais replied:
Oh, it seems two reviews have the same NR number, and maybe that's why it's happening.

I don't know if duplicate NRs are valid (eg, it could be for a rejected app and then for a valid app), so if possible let's support this scenario for now.

severinbeauvais replied:
This is now fixed. Resolving.

severinbeauvais commented:
Arwen, do you have any test data to show all 4 warning icon scenarios?

If you prepare which reviews and NRs to update, I can try to help with changing them.

ArwenQin commented:
> > Updated, made the NR number, date and identifier column narrower. Updated the redirect to NameX, will open in a new tab.
> 
> Looks great!
> 
> Since there's an 'X' to clear the dates and Status, what do you think of adding an 'X' to clear the NR Number, Identifying Number and Completing Party?
> 
> ![image](https://private-user-images.githubusercontent.com/10002889/360599629-76cfb244-1f61-4527-ad88-df638fe35324.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzNTQ5MjIsIm5iZiI6MTcyNDM1NDYyMiwicGF0aCI6Ii8xMDAwMjg4OS8zNjA1OTk2MjktNzZjZmIyNDQtMWY2MS00NTI3LWFkODgtZGY2MzhmZTM1MzI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIyVDE5MjM0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZmZTQ4YjI0NWZlZjQ3NGQ5Y2VmNTUzZTcxNTc4ODZiZDM2MWMwYzljM2YxNmRhMzJlNjdkMzQ5MDYxZmIxOTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ushSKL3FkbEeaBRGb1av-JB1NrgzbMPRJWQa5kLB2pg)

Agree. I will add the icons and pull up them

ArwenQin commented:
> Arwen, do you have any test data to show all 4 warning icon scenarios?
> 
> If you prepare which reviews and NRs to update, I can try to help with changing them.

I mocked the item.effectiveDate properties in the vue. I'm wondering could you please add some properties in the database?

severinbeauvais commented:
> Do you mean "open Name Request" action? I saw a Note "that NRs for Approved/Rejected/Abandoned reviews will not have an expiry date or id or phone/email, so do not show the "open name request" action for these ones."

Yes, the Open Name Request action.

I'm not entirely sure why I mentioned "expiry date" in that comment. As for the id or phone/email, that's left over from when we wanted to open the NR in Namerequest UI. Now that we open them in Name Examination UI, those fields are not needed.

I think it can't hurt to provide the ability to open the NR in all cases, subject to what Janis/Mihai say :)

severinbeauvais commented:
> > Also, can you check what the dropdown menu is attached to? It obscures its action button so I have to click away from it to close it.
> It's attached to the second row header

That sounds right.

You know what? The Status dropdown in the MBR works like this, too, so just leave this one alone for now. No changes needed.

severinbeauvais commented:
> > If you prepare which reviews and NRs to update, I can try to help with changing them.
> 
> I mocked the item.effectiveDate properties in the vue. I'm wondering could you please add some properties in the database?

Sure. We can do it together. Just let me know which reviews and NRs to update.

ArwenQin commented:
> Name Examination UI

Thank you.
I prepared 2 testing reviews:
![image](https://github.com/user-attachments/assets/d823d39b-93d2-490a-b89a-749c2b78350a)
These are the last 2 items in the table.
Both are future effective, and the second one has a NR

severinbeauvais commented:
/gcbrun

severinbeauvais commented on auth-web/src/components/auth/staff/continuation-application/ContinuationApplicationTable.vue:
```diff
@@ -71,7 +71,7 @@
                   </IconTooltip>
                   <IconTooltip
-                    v-if="daysLeft(item.effectiveDate) && daysLeft(item.effectiveDate) <= 3"
+                    v-if="daysLeft(item.futureEffectiveDate) && daysLeft(item.futureEffectiveDate) <= 3"
```
This line computes daysLeft twice in some cases. Consider having a single computed instead.

ArwenQin replied:
Updated

ArwenQin commented:
- Added close icon to all search fields, moved them up 2 px
 - Fixed the bug: calendar icon can't open the date picker
 - Fixed the bug: action sub-menu opens multiple times
 The status close icon is added by the v-select property "clearable", it seems I can't change its position, is it ok?
![image](https://github.com/user-attachments/assets/a620591c-5968-43d6-b724-b9f9345fff3a)

severinbeauvais commented on auth-web/src/components/auth/staff/continuation-application/ContinuationApplicationTable.vue:
```diff
@@ -82,8 +82,8 @@
-                        {{ daysLeft(item.effectiveDate) }}
-                        {{ daysLeft(item.effectiveDate) === 1 ? 'day' : 'days' }}.
+                        {{ daysLeft(item.futureEffectiveDate) }}
+                        {{ daysLeft(item.futureEffectiveDate) === 1 ? 'day' : 'days' }}.
```
Similar comment to above. A new computed could return "x days".

ArwenQin replied:
Updated, using daysLeftText()

bcregistry-sre commented:
Temporary Url for review: https://bcregistry-account-dev--pr-2978-f8g8fzgr.web.app

severinbeauvais commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://bcregistry-account-dev--pr-2978-f8g8fzgr.web.app

severinbeauvais commented:
![image](https://github.com/user-attachments/assets/5a72a072-3d94-4b6e-88a4-633f3fa4bad1)

This looks very good. Ask Janis if we should show hours instead of days if it's less than, say, 1 day.

NRs always expire at 11:59 pm on their last day, so we probably don't need to show hours for them. What does the tooltip say if the NR expires today?

Sample NR info:
![image](https://github.com/user-attachments/assets/df35399b-cd8b-46af-8282-8a91254981b1)

severinbeauvais commented:
### Test results

#### With NR expiring in 6 days (2024-08-29):

![image](https://github.com/user-attachments/assets/032541e5-d205-496c-ad72-d950203c057e)

#### With NR expiring today (2024-08-23):

![image](https://github.com/user-attachments/assets/85e453ae-ef2c-4e2f-9777-9eeb703bc601)

![image](https://github.com/user-attachments/assets/59626f15-bb78-4280-b78e-124276c0b4ae)

Do you think this is OK, or should it show "expiring today"? And also, why isn't the icon red?

ArwenQin commented:
> ### Test results
> #### With NR expiring in 6 days (2024-08-29):
> ![image](https://private-user-images.githubusercontent.com/10002889/361004890-032541e5-d205-496c-ad72-d950203c057e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ0MzI5MTksIm5iZiI6MTcyNDQzMjYxOSwicGF0aCI6Ii8xMDAwMjg4OS8zNjEwMDQ4OTAtMDMyNTQxZTUtZDIwNS00OTZjLWFkNzItZDk1MDIwM2MwNTdlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIzVDE3MDMzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE2Zjg1ZTg4ZjBhYjJjY2VkMzNjMWVkMzYyOTgxZDU2MzRlY2IyMzg0M2JkNjU5N2U5Mzk4ZjMyYTdhY2QwYjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qbODpFClaB97Uhsqg3pHV0PPGOSYs-3AedO1OTeDqxk)
> 
> #### With NR expiring today (2024-08-23):
> ![image](https://private-user-images.githubusercontent.com/10002889/361005339-85e453ae-ef2c-4e2f-9777-9eeb703bc601.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ0MzI5MTksIm5iZiI6MTcyNDQzMjYxOSwicGF0aCI6Ii8xMDAwMjg4OS8zNjEwMDUzMzktODVlNDUzYWUtZWYyYy00ZTJmLTk3NzctOWVlYjcwM2JjNjAxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIzVDE3MDMzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU0OTRhNjA3ZTJkNWU1NjUwYjA1ZjkyMGRiY2QyNmM3ZmNmNDRhMmY1Y2YyYmIxMmFhMDMzNmE5ZDc5ODNlZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.SSryPFwl0yyyLEOE0fSvRf_uF50RN4jyNuXfj1vE-xQ)
> 
> ![image](https://private-user-images.githubusercontent.com/10002889/361005456-59626f15-bb78-4280-b78e-124276c0b4ae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ0MzI5MTksIm5iZiI6MTcyNDQzMjYxOSwicGF0aCI6Ii8xMDAwMjg4OS8zNjEwMDU0NTYtNTk2MjZmMTUtYmI3OC00MjgwLWI3OGUtMTI0Mjc2YzBiNGFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIzVDE3MDMzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMyZDE3ZWQxNWRjMTVhMDViODQ0ZDkyM2NhMGIwZGI2NWEwZmYyNzVlMzI0ZmRhZjc3Y2Y4MzJiN2FmNTUyYmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.deiv7DLrFhDcCz9L3Q3Z11qmlJfwfc96w93iU40Gdec)
> 
> Do you think this is OK, or should it show "expiring today"? And also, why isn't the icon red?

Yes, if the NR expires in less than 24 hour, it will say "expires in 1 day". I think it's ok. Should I check with Janis?
I will check with Janis for the FED less than 24 hours.
The icon is red when the NR already expires (Figma below):
![image](https://github.com/user-attachments/assets/56486b17-0f88-45de-81f5-dbb6e8c4f4d7)

severinbeauvais commented:
> Yes, if the NR expires in less than 24 hour, it will say "expires in 1 day". I think it's ok. Should I check with Janis? I will check with Janis for the FED less than 24 hours. The icon is red when the NR already expires

Yes, check with Janis. As a possible solution, how about if the dates have their own tooltip like this?

![image](https://github.com/user-attachments/assets/d6049f7f-b65a-4551-8aef-6b66d8df9625)

Yes, you are correct about the red NR icon 👍

ArwenQin commented:
> > Yes, if the NR expires in less than 24 hour, it will say "expires in 1 day". I think it's ok. Should I check with Janis? I will check with Janis for the FED less than 24 hours. The icon is red when the NR already expires
> 
> Yes, check with Janis. As a possible solution, how about if the dates have their own tooltip like this?
> 
> ![image](https://private-user-images.githubusercontent.com/10002889/361019669-d6049f7f-b65a-4551-8aef-6b66d8df9625.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ0MzU2OTIsIm5iZiI6MTcyNDQzNTM5MiwicGF0aCI6Ii8xMDAwMjg4OS8zNjEwMTk2NjktZDYwNDlmN2YtYjY1YS00NTUxLThhZWYtNmI2NmQ4ZGY5NjI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIzVDE3NDk1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUxYmUyOGE0MzIyODFlNjgxYjYyMmFkN2M2MjdhZjg2MmFlNDUzODAxODVkNWIxZGE4ZTY5YzJhYmUzNTM5MzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KGoU4_j78ktOR52ele-ieDdHexJVnrskJJcEGPzhZL4)
> 
> Yes, you are correct about the red NR icon 👍

Sure, checking with Janis and cc you there.

ArwenQin commented:
Updated:
- All statuses can open NR
![image](https://github.com/user-attachments/assets/c2f6ee55-5239-4fbd-b78e-0a6c92d5121d)

- Updated the FED tooltip to show the full date and time
![image](https://github.com/user-attachments/assets/b9583b42-1968-4862-93e9-d4d453658342)

- Updated the NR date tooltip expiring today
![image](https://github.com/user-attachments/assets/f361e132-e192-438c-8ca2-c7de3e5e82e0)

severinbeauvais commented:
/gcbrun

severinbeauvais commented on auth-web/src/components/auth/staff/continuation-application/ContinuationApplicationTable.vue:
```diff
@@ -37,11 +48,95 @@
+              <!-- Displaying Formatted Future Effective Date and Tooltips-->
+              <template #[`item.futureEffectiveDate`]="{ item }">
+                <div>
+                  <v-tooltip bottom>
```
Can you use the same tooltip as elsewhere (which appears above the item and has a little pointer)?

severinbeauvais replied:
I see why you did this: it's to show the icon tooltip above and the date tooltip below. That's OK with me.

If you keep this, I'd like to see the little arrow, ie:

![image](https://github.com/user-attachments/assets/0b7ef5d6-c954-425c-bbd3-f0b392eece1b)

ArwenQin replied:
Sure, I will update it

severinbeauvais commented on auth-web/src/components/auth/staff/continuation-application/ContinuationApplicationTable.vue:
```diff
@@ -37,11 +48,95 @@
+                    </div>
+                  </IconTooltip>
+                  <IconTooltip
+                    v-if="item.futureEffectiveDate && daysLeft(item.futureEffectiveDate) <= 3"
```
Should we always show the icon and tooltip for a FE date?

ArwenQin replied:
Per Figma designs, we only show the icon and tooltip when the FE date passed, and when it's in 3 days, I think it makes sense?

severinbeauvais replied:
OK. Leave it. Resolving.

bcregistry-sre commented:
Temporary Url for review: https://bcregistry-account-dev--pr-2978-f8g8fzgr.web.app

severinbeauvais commented on auth-web/src/components/auth/staff/continuation-application/ContinuationApplicationTable.vue:
```diff
@@ -68,7 +163,12 @@
                       :scope="getIndexedTag('find-header-col', i)"
-                      class="font-weight-bold"
+                      :class="{
+                        'font-weight-bold': true,
```
No need to change this but FYI: since this class is always set, you can use the `class` prop on this element. (Ie, you can use both `class` and `:class`.)

severinbeauvais commented on auth-web/src/components/auth/staff/continuation-application/ContinuationApplicationTable.vue:
```diff
@@ -190,13 +375,41 @@
                     <v-btn
                       color="primary"
-                      :class="['open-action-btn', getButtonLabel(item.status).toLowerCase()]"
+                      :class="['open-action-btn', { active: item.nrNumber }]"
```
I don't understand this syntax. What does the right hand side evaluate to? Should be True or False.

ArwenQin replied:
When the item has a nrNumber, We will adjust the styles of the "Open" Button:
![image](https://github.com/user-attachments/assets/999451c6-9b5a-4752-9511-41a9d2a78bf6)

severinbeauvais replied:
Please change this to the usual syntax:
```
   class="open-action-btn"
   :class="{ 'active': item.nrNumber }"
```

ArwenQin replied:
Updated

severinbeauvais replied:
Looks good. Resolving.

severinbeauvais commented on auth-web/src/components/auth/staff/continuation-application/ContinuationApplicationTable.vue:
```diff
@@ -405,8 +657,58 @@ export default defineComponent({
+    const feDaysLeftText = (effectiveDate: string): string => {
+      const diffHours = moment(effectiveDate).diff(moment(), 'hours')
+      if (diffHours > 0 && diffHours <= 24) {
+        return '1 day'
```
I think this should show the number of hours left if it's less than 1 day.

ArwenQin replied:
Will update it

ArwenQin replied:
Updated:
![image](https://github.com/user-attachments/assets/802e7036-c6a8-40ad-abb9-80e697a1cb23)

severinbeauvais replied:
Nice! Resolving.

severinbeauvais commented on auth-web/src/components/auth/staff/continuation-application/ContinuationApplicationTable.vue:
```diff
@@ -405,8 +657,58 @@ export default defineComponent({
+    // Method to display NR days left text
+    const nrDaysLeftText = (effectiveDate: string): string => {
+      const diffHours = moment(effectiveDate).diff(moment(), 'hours')
+      if (diffHours > 0 && diffHours <= 24) {
```
I think this assumes that the NR will expire at the end of the day. Would it better to check if the NR expiry day is the same as today?

ArwenQin replied:
Will update, thanks

ArwenQin replied:
Updated

severinbeauvais replied:
Looks good. Resolving.

ArwenQin commented on auth-web/src/components/auth/staff/continuation-application/ContinuationApplicationTable.vue:
```diff
@@ -37,11 +48,97 @@
+                      </span>
+                    </template>
+                    <div class="bottom-tooltip">
+                      <span>{{ formatDate(item.futureEffectiveDate) }} at {{ formatTime(item.futureEffectiveDate) }} Pacific time</span>
```
I try to show the little arrow on top of the tooltip, but I'm not sure why this class="bottom-tooltip" didn't work
![image](https://github.com/user-attachments/assets/9a72c066-071a-44d2-be67-7557143a1881)

(it's the same style used by the IconTooltip component)

severinbeauvais replied:
Try without `::v-deep`.

severinbeauvais replied:
We need to use v-deep when the HTML is rendered by a Vuetify component.

Here, it's "static" HTML in the template.

ArwenQin replied:
I tried without ::v-deep, but it didn't work either

ArwenQin replied:
It seems I couldn't change the styles of the tooltip

severinbeauvais replied:
I see that `:before` isn't in the DOM. Let me see if I can figure it out.

![image](https://github.com/user-attachments/assets/764cbbc0-b1f6-4ee2-a857-e0f819bd39fc)

severinbeauvais replied:
Found it. It's because your CSS is nested in the form. Move it outside and the v-deep will work.

PS You'll have to fix the colour of the pointer.

ArwenQin replied:
Thanks a lot! Fixed it and fixed the color:
![image](https://github.com/user-attachments/assets/313c0d8d-d12c-40b9-854a-385e9db23d00)

severinbeauvais replied:
Looks great! 🎉

severinbeauvais commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://bcregistry-account-dev--pr-2978-f8g8fzgr.web.app

severinbeauvais commented on auth-web/package-lock.json:
```diff
@@ -58,7 +58,7 @@
         "@typescript-eslint/eslint-plugin": "^6.4.0",
         "@typescript-eslint/parser": "^6.4.0",
-        "@volar-plugins/vetur": "*",
+        "@volar-plugins/vetur": "latest",
```
Please undo this change.

Also, please update the app version again.

Then, go ahead and merge this PR.

ArwenQin replied:
Thanks Sev! I just undo this and updated the app version again.
Could you please merge it?
Thank you.
